oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,92 @@ impl super::Node for WasmNode {
-    let (_, reader_handle) =
-        runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
-
-    // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
-    // Minimal module:
-    // \0asm - magic
-    // 0x01 - version (in little-endian)
-    let module =
-        wasmi::Module::from_buffer(vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]).unwrap();
-
-    let mut node = WasmNode::new(
-        "test",
-        runtime_ref,
-        TEST_NODE_ID,
-        Arc::new(module),
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use std::env;
+    use wat::parse_file;
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
+        // Minimal module:
+        // \0asm - magic
+        // 0x01 - version (in little-endian)
+        let module = wasmi::Module::from_buffer(buffer).unwrap();
+
+        Box::new(WasmNode::new(
+            "test",
+            runtime_ref,
+            TEST_NODE_ID,
+            Arc::new(module),
+            entrypoint.to_string(),
+            reader_handle,
+        ))
+    }
+
+    fn from_testfile(file_name: &str) -> Vec<u8> {
+        let mut path = env::current_dir().unwrap();
+        path.pop();
+        path.pop();
+        path.push("testdata");
+        path.push(file_name);
+        parse_file(path).unwrap()
+    }
+
+    #[test]
+    fn wasm_with_invalid_args_fails() {
+        let buffer = vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00];
+
+        // Wrong magic: The minimal module does not have the necessary entry point, so it should
+        // fail immediately.
+        let mut node = setup_node(buffer, "test_function");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_module_without_content_fails() {
+        let binary = from_testfile("empty.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_minimal_module_succeeds() {
+        let binary = from_testfile("minimal.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Ok(()), node.start());
+        node.stop();
```

Comment:
Does this make sense here? It causes the test to panic:
`thread '<unnamed>' panicked at 'remove_node_id: No such node_id', oak/server/rust/oak_runtime/src/runtime/mod.rs:585:28`
And why does not the test fail!

Reply:
Ahh the panic is because `remove_node_id` is called by the node when stopping, but `remove_node_id` doesn't filter for RUNTIME_NODE_ID/TEST_NODE_ID, which it should do. Can you add it is similar to https://github.com/project-oak/oak/blob/a005757bd99fe6642e3813108283d28e7fc42f61/oak/server/rust/oak_runtime/src/runtime/mod.rs#L254 ? Note that RUNTIME_NODE_ID/TEST_NODE_ID are the same value.